go/vt/vtgate/planbuilder/operators/ast2op.go

```diff
@@ -215,6 +449,133 @@ func createOperatorFromDelete(ctx *plancontext.PlanningContext, deleteStmt *sqlp
 		return nil, err
 	}
 
+	delClone := sqlparser.CloneRefOfDelete(deleteStmt)
+	// Create the delete operator first.
+	delOp, err := createDeleteOperator(ctx, deleteStmt, qt, vindexTable, routing)
+	if err != nil {
+		return nil, err
+	}
+
+	// Now we check for the foreign key mode and make changes if required.
+	ksMode, err := ctx.VSchema.ForeignKeyMode(vindexTable.Keyspace.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmanaged foreign-key-mode, we don't need to do anything.
+	if ksMode != vschemapb.Keyspace_FK_MANAGED {
+		return delOp, nil
+	}
+	childFks := vindexTable.ChildFKsNeedsHandling(vindexes.DeleteAction)
+	// If there are no foreign key constraints, then we don't need to do anything.
+	if len(childFks) == 0 {
+		return delOp, nil
+	}
+	// If the delete statement has a limit, we don't support it yet.
+	if deleteStmt.Limit != nil {
+		return nil, vterrors.VT12001("foreign keys management at vitess with limit")
+	}
+
+	var fkChildren []*FkChild
+	var selectExprs []sqlparser.SelectExpr
+	for _, fk := range childFks {
```

Comment:
more deep nesting. my rule of thumb if that switch cases should not contain a lot of code. a single method statement, and maybe error check, but nothing else. this is all highly subjective, but it's definitely the pattern used in the operators package